src/components/CustomDisciplineMenuItem.tsx

```diff
@@ -36,21 +44,78 @@ const ErrorText = styled.Text`
   margin: 10px;
 `
 
+const DeleteContainer = styled.View`
+  width: ${widthOfTrashButton};
+`
+
+const DeleteButton = styled(RectButton)`
+  height: ${wp('21%')}px;
+  margin: 0 16px 8px -16px;
+  align-items: center;
+  flex: 1;
+  justify-content: center;
+  background-color: ${props => props.theme.colors.lunesFunctionalIncorrectDark};
+  width: ${widthOfTrashButton};
+`
+
 interface CustomDisciplineMenuItemPropsType {
   apiKey: string
   selectedId: string | null
-  setSelectedId: (input: string) => void
+  setSelectedId: React.Dispatch<React.SetStateAction<string | null>>
   navigation: StackNavigationProp<RoutesParamsType, 'Home'>
+  refresh: () => void
 }
 
 const CustomDisciplineMenuItem = ({
   apiKey,
   selectedId,
   setSelectedId,
-  navigation
+  navigation,
+  refresh
 }: CustomDisciplineMenuItemPropsType): JSX.Element => {
   const { data, loading } = useLoadGroupInfo(apiKey)
 
+  const [isModalVisible, setIsModalVisible] = useState<boolean>(false)
+
+  const renderRightAction = (progress: Animated.AnimatedInterpolation): ReactElement => {
+    const trans = progress.interpolate({
+      inputRange: [0, 1],
+      outputRange: [50, 0]
+    })
+
+    const showConfirmationModal = (): void => {
+      close()
+      setIsModalVisible(true)
+    }
+
+    return (
+      <DeleteContainer>
+        <Animated.View style={{ flex: 1, transform: [{ translateX: trans }] }}>
+          <DeleteButton onPress={showConfirmationModal}>
+            <TrashBinIcon testID={'trash-bin-icon'} />
+          </DeleteButton>
+        </Animated.View>
+      </DeleteContainer>
+    )
+  }
+
+  const deleteModule = (): void => {
+    AsyncStorage.deleteCustomDiscipline(apiKey)
+      .then(() => refresh())
+      .catch((e: Error) => console.log(e.message))
+    setIsModalVisible(false)
+  }
+
+  let swipeableRow: Swipeable
+
+  const updateRef = (ref: Swipeable): void => {
+    swipeableRow = ref
+    swipeableRow?.close()
```

Comment:
```suggestion
    swipeableRow.close()
```

Reply:
no this is needed, when opening the screen the ref can be null.